CMakeLists.txt

```diff
@@ -100,6 +100,12 @@ ENDIF()
 INCLUDE(cmake/Conan.cmake)
 RUN_CONAN()
 
+ADD_SUBDIRECTORY(src)
+
+IF(CMAKE_CROSSCOMPILING)
+    RETURN()
```

Comment:
why should _building_ of tests, especially constexpr tests, be disabled for cross compilation?

Is there a more fine-tuned approach, for inside the test and fuzz_test directories?

if we really can not build tests, I'd prefer a message and disabling the test builds instead of the premature return in this file.

Reply:
I have added a message here. My idea was here to return as early as possible because at the moment it doesn't make any sense to try to compile test with gtest / catch2 which aren't executable on the device itself.